openedx/core/djangoapps/programs/utils.py

```diff
@@ -75,7 +75,10 @@ def __init__(self, user, enrollments=None, uuid=None):
         for enrollment in self.enrollments:
             # enrollment.course_id is really a CourseKey (╯ಠ_ಠ）╯︵ ┻━┻
             enrollment_id = unicode(enrollment.course_id)
-            self.enrolled_run_modes[enrollment_id] = enrollment.mode
+            mode = enrollment.mode.replace(
+                CourseMode.NO_ID_PROFESSIONAL_MODE, CourseMode.PROFESSIONAL
```

Comment:
This is an unlikely edge case, but say someone running edx-platform (not necessarily at edX) decided to add a `no-id-professional-with-rice` mode. The replacement here would still occur even though that may not be desired. A more surgical approach (e.g., checking equality, like [this](https://github.com/edx/edx-platform/pull/14931/files#diff-1072896d9c36e00edfdd4998d9e13d22R309)) could prevent unexpected behavior.